e2e/test/scenarios/visualizations-tabular/pivot_tables.cy.spec.js

```diff
@@ -1272,6 +1272,65 @@ describe("scenarios > visualizations > pivot tables", { tags: "@slow" }, () => {
       cy.findByTestId("qb-save-button").should("have.attr", "data-disabled");
     });
   });
+
+  describe("issue 38265", () => {
+    beforeEach(() => {
+      createQuestion(
+        {
+          query: {
+            "source-table": ORDERS_ID,
+            aggregation: [
+              ["count"],
+              [
+                "sum",
+                ["field", ORDERS.SUBTOTAL, { "base-type": "type/Float" }],
+              ],
+            ],
+            breakout: [
+              [
+                "field",
+                ORDERS.CREATED_AT,
+                { "base-type": "type/DateTime", "temporal-unit": "month" },
+              ],
+              [
+                "field",
+                PEOPLE.STATE,
+                { "base-type": "type/Text", "source-field": ORDERS.USER_ID },
+              ],
+              [
+                "field",
+                PRODUCTS.CATEGORY,
+                { "base-type": "type/Text", "source-field": ORDERS.PRODUCT_ID },
+              ],
+            ],
+          },
+          display: "pivot",
+        },
+        {
+          wrapId: true,
+          idAlias: "questionId",
+        },
+      );
+    });
+
+    it("correctly filters the query when zooming in on a **row** header (metabase#38265)", () => {
+      cy.signInAsNormalUser();
+      visitQuestion("@questionId");
+      cy.findByTestId("pivot-table").findByText("KS").click();
+      popover().findByText("Zoom in").click();
+
+      cy.log("Filter pills");
+      const filters = cy.findAllByTestId("filter-pill");
+      filters.should("have.length", 1);
+      filters.first().should("have.text", "User → State is KS");
```

Comment:
Never use `const` like this in Cypress.
It doesn't work the way you'd expect it to.
https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Return-Values

In this particular case it's perfectly fine to do the following, because the assertion will fail anyway if it finds more than one element with this test-id.
```suggestion
      cy.findByTestId("filter-pill").should("have.text", "User → State is KS");
```